sdk/core/src/request.rs

```diff
@@ -0,0 +1,60 @@
+use http::{HeaderMap, Method};
+use std::fmt::Debug;
+use url::Url;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Request {
+    pub url: Url,
+    pub method: Method,
+    pub headers: HeaderMap,
+    pub payload: bytes::Bytes,
```

Comment:
I would rather request be completely opaque. This helps with future compatibility. Can we just make these getters and setters instead?

Reply:
Yes I agree. What do you think about the `getset` crate?

Reply:
I'd rather avoid a dependency if we can. Let's write it out and see how hard it is to maintain. It would be trivial to write a macro_rules macro for this case, if we really need it.